17.0 event compatibility layer sri

stefanorigano opened the pull request:
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/154013).

reth-odoo commented on addons/website_event_migration_layer/static/src/js/website_event_track.js:
```diff
@@ -0,0 +1,62 @@
+                });
+
+                window.addEventListener("resize", () => {
+                    this._computeAgendasSize();
```
And here just pass in "agendas" instead of using "this" inside the method.

Makes the flow easier to follow

reth-odoo commented on addons/website_event_track/static/src/scss/event_track_templates.scss:
```diff
@@ -150,6 +178,32 @@
     }
 }
 
+@if $o-wevent-migration-layer {
+    .o_we_online_agenda_overlay {
+        @include o-position-absolute(0, 0, 63px);
+        background-image: linear-gradient(90deg, rgba($body-color, 0), rgba($body-color, 0.1), rgba($body-color, 0.3));
+        opacity: 0;
+        transition: opacity .1s ease;
+    }
+
+    .o_we_online_agenda_has_scroll {
+        border-top-color: $border-color;
+
+        + .o_we_online_agenda_overlay {
+            opacity: 1;
+            transition: opacity .5s ease;
+        }
+    }
+
+    .o_we_online_agenda_has_scroll.o_we_online_agenda_is_scrolling + .o_we_online_agenda_overlay {
+        opacity: .5;
+    }
+
+    .o_we_track_day_header {
+        z-index: 2;
+    }
+}
+
```
Ah I get it now, I thought the overlay was going to be more visible, that's why I asked if it was done. It works quite nicely actually

reth-odoo commented on addons/website_event_migration_layer/static/src/js/website_event_track.js:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * @private
+     */
+    _computeAgendasSize: function () {
```
would rather call it "updateScrollOverlay" or something. It doesn't really compute any size.

reth-odoo commented on addons/website_event_migration_layer/views/website_event.xml:
```diff
@@ -0,0 +1,414 @@
+<!-- addons/website_event/views/event_templates_page.xml -->
+<!-- =================================================== -->
+<template id="navbar__ml" inherit_id="website_event.navbar" name="navbar__ml">
+    <xpath expr="//div" position="replace"/>
```
Can we be a bit more specific about the divs? Even if it works, it seems weird to have the same selector twice

reth-odoo reviewed (commented):
Some comments, it's quite confusing to review now but we have to do what we have to do.

Overall it seems fine, I think functional testing will help more than the review for most issues if any.

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+    </xpath>
+    <xpath expr="//t[@t-as='booth_category']//label" position="attributes">
+        <attribute name="t-attf-class"/>
+        <attribute name="t-attf-class">d-block h-100 #{'o_wbooth_category_unavailable overflow-hidden' if booth_category_unavailable else ''}"</attribute>
```
Can be changed directly in base module

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+        <attribute name="t-attf-class">d-block h-100 #{'o_wbooth_category_unavailable overflow-hidden' if booth_category_unavailable else ''}"</attribute>
+    </xpath>
+    <xpath expr="//t[@t-as='booth_category']//div[hasclass('card-body')]" position="attributes">
+        <attribute name="class" add="d-flex flex-wrap w-100 gap-2 justify-content-between flex-grow-0 pb-0" separator=" "/>
```
Would say ok, not changing DOM

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+        <attribute name="class" add="d-flex flex-wrap w-100 gap-2 justify-content-between flex-grow-0 pb-0" separator=" "/>
+    </xpath>
+    <xpath expr="//t[@t-as='booth_category']//h5" position="attributes">
+        <attribute name="class" add="my-0" separator=" "/>
```
Can be changed directly in base module

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+    <xpath expr="//t[@t-as='booth_category']//h5" position="attributes">
+        <attribute name="class" add="my-0" separator=" "/>
+    </xpath>
+    <xpath expr="//t[@t-as='booth_category']//div[@t-field='booth_category.description']" position="replace"/>
+    <xpath expr="//t[@t-as='booth_category']//div[hasclass('card-body')]" position="after">
+        <div class="w-100 small" t-attf-id="o_wbooth_booth_description_#{booth_category.id}" t-field="booth_category.description"/>
+    </xpath>
```
Really no other way to do it than by changing DOM ?

stefanorigano replied:
Unfortunately not :/

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+        <div class="w-100 small" t-attf-id="o_wbooth_booth_description_#{booth_category.id}" t-field="booth_category.description"/>
+    </xpath>
+    <xpath expr="//t[@t-as='booth_category']//div[@t-if='booth_category_unavailable']" position="attributes">
+        <attribute name="class" add="text-bg-danger" remove="bg-danger" separator=" "/>
```
Can be done in base module

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+    </xpath>
+
+    <xpath expr="//t[@t-if='available_booth_category_ids']//h4" position="replace">
+        <h5 class="my-0">Location</h5>
```
Same: display-h5 or something similar in base module ?

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+        <attribute name="class" remove="flex-column align-content-end border-top" add="align-items-center gap-2 mt-4 mb-5" separator=" "/>
+    </xpath>
+    <xpath expr="//button[hasclass('o_wbooth_registration_submit')]" position="attributes">
+        <attribute name="class" remove="ms-auto fw-bold" separator=" "/>
```
Can probably be done in base module

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+        </div>
+    </xpath>
+    <xpath expr="//form[hasclass('js_website_submit_form')]" position="attributes">
+        <attribute name="class" add="col-lg-9" separator=" "/>
```
Can be done in base module

tde-banana-odoo commented on addons/website_event_migration_layer/views/website_event_booth.xml:
```diff
@@ -0,0 +1,104 @@
+    </xpath>
+    <!-- Contact Details -->
+    <xpath expr="//h4" position="replace">
+        <h5 class="my-3">Contact Details</h5>
```
Same proposal to change display in stable in base module

tde-banana-odoo reviewed (commented):
First check for overrides of somr modules (booth, booth_sale, event_sale, meet_quiz) as we could probably lessen the extra layer and move "easy parts" back to base module.

Sorry as it implies some back and forth, but actually seeing it as a collection of xpaths helps a lot seeing the change, grepping in 17.0 and trying to see if it will break something or not :) .

Will do a pass for other layers soon (more to check ^^)

stefanorigano commented:
@reth-odoo and @tde-banana-odoo ,  thanks for your reviews, I reworked the PR as requested. 

The resulting number of sub-templates is down from 8 to 5 now.

About "toggle classes operations", I've moved 99% of them in default templates and wrote proper commit messages.
The remaining 1% remains in the new sub-modules;  these changes make sense when sub-modules (and their new templates) are active only indeed. Same for the SCSS, part of the new code is triggered if sub-modules are active only.

Tested successfully in the following scenario:
- 17.0 db: apply fixes → lunch **without** `-u`
- 17.0 db: apply fixes → lunch **with** `-u`
- 17.0 db: apply fixes **+** activate all sub-modules

**Important**: I'm not able to handle JS tests in sub-modules (tried multiple approaches, nothing works, not exactly my field). 

Despite I'm eager to learn more, it's probably not the right time considering the 10M€/d constrain😅
I'd suggest to fixup your black-magic on top 🙌
Thanks

reth-odoo reviewed (commented):
Still not much to say on my part, it's definitely a lot easier to understand what's going on with the current separation of things. I think it's as good as it gets now :)
Selectors look ok to me.

For the one test failing I won't add a commit since it's a very simple fix. Just using `a.btn-primary[href$="/register"]` instead of `a.btn-primary:contains("Register")` in `test_event_full/` seems to work fine for both the patched and unpatched templates. Seems reasonable :)

stefanorigano commented:
@reth-odoo , test updated, green on runbot (except migration stuff). Thanks 👍

stefanorigano commented:
Thanks for your time 👌
Closing this in favour of https://github.com/odoo/odoo/pull/152434 and  https://github.com/odoo/odoo/pull/156250
